src/main/scala/chisel3/tester/BackendInterface.scala

```diff
@@ -90,3 +90,15 @@ trait TestEnvInterface {
     */
   def checkpoint()
 }
+
+trait ScalaSimulator
```

Comment:
Good question, in some ways it seem like it might be nice to plug in other experimental 
scala based simulations.  This is PR is sort of a trial balloon on how to integrate treadle.
I'll talk to Richard about this.

Reply:
If this trait is empty (and not being used as an abstract base class) I don't think there's a need for it.